NachoSpigot-Server/src/main/java/net/minecraft/server/Entity.java

```diff
@@ -1181,11 +1184,23 @@ public double f(double d0, double d1, double d2) {
     }
 
     public double h(Entity entity) {
-        double d0 = this.locX - entity.locX;
-        double d1 = this.locY - entity.locY;
-        double d2 = this.locZ - entity.locZ;
+        if (entity instanceof EntityPlayer && NachoConfig.enableImprovedHitReg) {
```

Comment:
Boolean compare is less expensive than the `instanceof` operator. (thank you for noticing @demkom58!)
```suggestion
        if (NachoConfig.enableImprovedHitReg && entity instanceof EntityPlayer) {
```